client_servers.go

```diff
@@ -0,0 +1,705 @@
+	Identifier  string `json:"identifier"`
+	UUID        string `json:"uuid"`
+	InternalID  int    `json:"internal_id"`
+	Name        string `json:"name"`
+	Node        string `json:"node"`
+	SFTP        struct {
+		IP   string `json:"ip"`
+		Port int64  `json:"port"`
+	} `json:"sftp_details"`
+	Description   string        `json:"description"`
+	Limits        Limits        `json:"limits"`
+	Invocation    string        `json:"invocation"`
+	DockerImage   string        `json:"docker_image"`
+	EggFeatures   []string      `json:"egg_features"`
+	FeatureLimits FeatureLimits `json:"feature_limits"`
+	Status        string        `json:"status"`
+	Suspended     bool          `json:"is_suspended"`
+	Installing    bool          `json:"is_installing"`
+	Transferring  bool          `json:"is_transferring"`
+}
+
+func (c *Client) Servers() ([]*ClientServer, error) {
+	req := c.newRequest("GET", "", nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Data []struct {
+			Attributes *ClientServer `json:"attributes"`
+		} `json:"data"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	servers := make([]*ClientServer, 0, len(model.Data))
+	for _, s := range model.Data {
+		servers = append(servers, s.Attributes)
+	}
+
+	return servers, nil
+}
+
+func (c *Client) Server(identifier string) (*ClientServer, error) {
+	req := c.newRequest("GET", "/servers/"+identifier, nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Attributes ClientServer `json:"attributes"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	return &model.Attributes, err
+}
+
+type WebSocketAuth struct {
+	Socket string `json:"socket"`
+	Token  string `json:"token"`
+}
+
+func (c *Client) ServerWebSocket(identifier string) (*WebSocketAuth, error) {
+	req := c.newRequest("GET", fmt.Sprintf("/servers/%s/websocket", identifier), nil)
+	res, err := c.Http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := validate(res)
+	if err != nil {
+		return nil, err
+	}
+
+	var model struct {
+		Data WebSocketAuth `json:"data"`
+	}
+	if err = json.Unmarshal(buf, &model); err != nil {
+		return nil, err
+	}
+
+	return &model.Data, nil
+}
+
+type Resources struct {
+	MemoryBytes    int64 `json:"memory_bytes"`
+	DiskBytes      int64 `json:"disk_bytes"`
+	CPUAbsolute    int64 `json:"cpu_absolute"`
+	NetworkRxBytes int64 `json:"network_rx_bytes"`
+	NetworkTxBytes int64 `json:"network_tx_bytes"`
+	Uptime         int64 `json:"uptime"`
+}
+
+type Stats struct {
+	State     string    `json:"current_state,omitempty"`
+	Suspended bool      `json:"is_suspended"`
+	Resources Resources `json:"resources"`
+}
+
+func (c *Client) ServerStatistics(identifier string) (*Stats, error) {
```

Comment:
I almost want this to be `ServerResources` instead of `Statistics` 

I also feel like the funcs to get information should be `Get{RESOURCE}` i.e. `GetServerStatistics`. I understanding not having get in front of client functions but they have items, such as variables, they can set as well.

Reply:
> I almost want this to be `ServerResources` instead of `Statistics`

That would also mean renaming the `Stats` struct to something else, or perhaps have the `Resources` struct as a nested object?

 > I also feel like the funcs to get information should be `Get{RESOURCE}` i.e. `GetServerStatistics`. I understanding not having get in front of client functions but they have items, such as variables, they can set as well.

Yeah I can change it for some of them, not sure about this one though.

Reply:
I see now that you have 2 structs and calling one as nested object. That makes sense but the panel api endpoint is resources.

Maybe `Resources` and `ResourceUsage` Instead of `Stats` and `Resources`

Reply:
I went with `ResourceUsage` and `Resources` 👍